pkg/okteto/kubetoken_client.go

```diff
@@ -14,56 +14,92 @@
 package okteto
 
 import (
+	"encoding/json"
+	"errors"
 	"fmt"
+	"github.com/okteto/okteto/pkg/types"
 	"io"
 	"net/http"
+	"net/url"
+)
 
-	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+const (
+	// kubetokenPathTemplate (baseURL, namespace)
+	kubetokenPathTemplate = "%s/auth/kubetoken/%s"
 )
 
-const kubetokenPath = "auth/kubetoken"
+var (
+	errRequest               = errors.New("failed request")
+	errStatus                = errors.New("status error")
+	errUnauthorized          = errors.New("unauthorized")
+	errKubetokenNotAvailable = errors.New("kubetoken service not found")
+)
 
-type KubeTokenClient struct {
-	httpClient  *http.Client
-	url         string
-	contextName string
+type kubeTokenClient struct {
+	httpClient *http.Client
 }
 
-func NewKubeTokenClient(contextName, token, namespace string) (*KubeTokenClient, error) {
-	if contextName == "" {
-		return nil, oktetoErrors.ErrCtxNotSet
+func newKubeTokenClient(httpClient *http.Client) *kubeTokenClient {
+	return &kubeTokenClient{
+		httpClient: httpClient,
 	}
+}
 
-	httpClient, url, err := newOktetoHttpClient(contextName, token, fmt.Sprintf("%s/%s", kubetokenPath, namespace))
+func getKubetokenURL(baseURL, namespace string) (*url.URL, error) {
+	return url.Parse(fmt.Sprintf(kubetokenPathTemplate, baseURL, namespace))
+}
+
+func (c *kubeTokenClient) GetKubeToken(baseURL, namespace string) (*types.KubeTokenResponse, error) {
+	url, err := getKubetokenURL(baseURL, namespace)
 	if err != nil {
 		return nil, err
 	}
 
-	return &KubeTokenClient{
-		httpClient:  httpClient,
-		url:         url,
-		contextName: contextName,
-	}, nil
-}
-
-func (c *KubeTokenClient) GetKubeToken() (string, error) {
-	resp, err := c.httpClient.Get(c.url)
+	resp, err := c.httpClient.Get(url.String())
 	if err != nil {
-		return "", fmt.Errorf("failed GET request: %w", err)
+		return nil, fmt.Errorf("GetKubeToken %w: %w", errRequest, err)
 	}
 
 	if resp.StatusCode == http.StatusUnauthorized {
-		return "", fmt.Errorf(oktetoErrors.ErrNotLogged, c.contextName)
+		return nil, fmt.Errorf("GetKubeToken %w", errUnauthorized)
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("GET request returned status %s", resp.Status)
+		return nil, fmt.Errorf("GetKubeToken %w: %s", errStatus, resp.Status)
 	}
 
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return "", fmt.Errorf("failed to read kubetoken response: %w", err)
+		return nil, fmt.Errorf("failed to read kubetoken response: %w", err)
+	}
+
+	var kubeTokenResponse types.KubeTokenResponse
+	err = json.Unmarshal(body, &kubeTokenResponse)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal kubetoken response: %w", err)
+	}
+
+	return &kubeTokenResponse, nil
+}
+
+func (c *kubeTokenClient) CheckService(baseURL, namespace string) error {
```

Comment:
Just to make sure. This function is only used on tests.Should we call it when we init the kubetoken client?

Reply:
this is going to be used in the following PRs, we added as both andrea and me will needed for our issues

Reply:
Thanks for explaining it :)